### PR TITLE
[webgpu] Add stridedSlice kernel

### PR DESCRIPTION
**Pull request by axinging**
FEATURE

Fix: https://github.com/tensorflow/tfjs/issues/2363

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2441)

### CODE REVIEW

**Comment by axinging** [comment]
@annxingyuan , PTAL

**Comment by annxingyuan** [comment]
Hi @axinging - is this ready for another look? Would you mind also merging master in and resolving conflicts? Thank you!

**Comment by axinging** [comment]
Done，@annxingyuan

File: tfjs-backend-webgpu/src/backend_webgpu.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@
 
 import './flags_webgpu';
 
-import {backend_util, DataStorage, DataType, engine, env, findBackend, KernelBackend, Rank, RecursiveArray, ShapeMap, Tensor, Tensor2D, Tensor3D, Tensor4D, TimingInfo, util} from '@tensorflow/tfjs-core';
+// TODO(xing.xu): use slice_util from backend_util:
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/backend_webgpu.ts]: Remove this TODO.

File: tfjs-backend-webgpu/src/kernels/strided_slice_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import {WebGPUProgram} from './webgpu_program';
+
+export class StridedSliceProgram implements WebGPUProgram {
+  variableNames = ['x'];
+  outputShape: number[];
+  userCode: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workPerThread = 1;
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/strided_slice_webgpu.ts]: Have you tried increasing the workPerThread? If not we can try it in a follow-up PR. Would you mind adding a TODO here to increase the workPerThread?

**Comment by annxingyuan** [comment]
@axinging two more tiny comments - almost there!

File: tfjs-backend-webgpu/src/kernels/strided_slice_webgpu.ts (hunk 2 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/strided_slice_webgpu.ts]: I didn't try it yet, I will try it in a followup PR.And TODO is added.

**Pull request merged**